### PR TITLE
Encoder enhancements and bug fixes

### PR DESCRIPTION
**Pull request by akshayragir833**
- Fix for oss-fuzz issue.

- Aligned fuzzer application with the test application

Tests done: Smoke test

### CODE REVIEW

**Review by divya-bm — APPROVED** [review]

**Review by sandeshvenkatesh — APPROVED** [review]

File: fuzzer/xaac_enc_fuzzer.cpp [hunk 1]
```diff
@@ -203,105 +203,16 @@ static VOID ixheaace_read_drc_config_params(
...
-        fuzzed_data->ConsumeBool();
-    if (1 == pstr_enc_loudness_info_set->str_loudness_info_album[n].true_peak_level_present) {
-      pstr_enc_loudness_info_set->str_loudness_info_album[n].true_peak_level =
-          fuzzed_data->ConsumeFloatingPoint<FLOAT32>();
-      pstr_enc_loudness_info_set->str_loudness_info_album[n].true_peak_level_measurement_system =
-          fuzzed_data->ConsumeIntegral<WORD8>();
-      pstr_enc_loudness_info_set->str_loudness_info_album[n].true_peak_level_reliability =
-          fuzzed_data->ConsumeIntegral<WORD8>();
-    }
+  pstr_uni_drc_config->str_channel_layout.layout_signaling_present = 0;
```
Comment (harishdm) [inline fuzzer/xaac_enc_fuzzer.cpp:206]: this change seems wrong.

with this drc path will not be tested well.

please revert this change.

Comment (harishdm) [reply fuzzer/xaac_enc_fuzzer.cpp:206]: @akshayragir833 @divya-bm @sandeshvenkatesh
